### PR TITLE
feat: add convenience methods for selector identifiers to ContractInstance

### PR DESCRIPTION
**Pull request by mikeshultz**
### What I did

Adds convenience method/props `selector_identifiers` and `identifier_lookup` to `ContractInstance` for easy selector/identifier lookups.

fixes: #1682

### How I did it

Pass through methods to the underlying `ContractType`

### How to verify it

See tests?

### Checklist

- [X] All changes are completed
- [X] New test cases have been added
- [x] Documentation has been updated

### CODE REVIEW

File: src/ape/contracts/base.py [hunk 1]
```diff
@@ -711,6 +711,14 @@ def poll_logs(
 class ContractTypeWrapper(ManagerAccessMixin):
     contract_type: ContractType
 
+    @property
+    def selector_identifiers(self) -> Dict[str, str]:
```
Comment (antazoey) [inline src/ape/contracts/base.py:715]: optional: doc strings for these

**Review by antazoey — DISMISSED** [review]
LGTM, one small optional thing

File: src/ape/contracts/base.py (hunk 1 above)
Comment (mikeshultz) [reply src/ape/contracts/base.py:715]: [8d6ef6c](https://github.com/ApeWorX/ape/pull/1870/commits/8d6ef6c90aedfb6839b20229e7fd5b348a52ca5a)

**Review by antazoey — APPROVED** [review]

**Comment by gosuto-inzasheru** [comment]
awesome work, thank you sir 🫡

**Pull request merged**